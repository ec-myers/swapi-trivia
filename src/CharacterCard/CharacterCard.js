import React from 'react';
import PropTypes from 'prop-types';
import './CharacterCard.scss';

const CharacterCard = ({character, toggleFavorite, favorites}) => {
  const { name, homeworld, population, species, films } = character;
  let isFavorite = favorites.map(char => char.name).includes(name) ? 'favorite' : '' ;
  let btnMessage = isFavorite ? 'Remove Favorite' : 'Add Favorite'
  const allFilms = films.map(film => {
  return <li key={film}>{film}</li>
  })

  return (
    <div className={`Character-Card ${isFavorite}`} >
      <h2>{name}</h2>
      <h3>Homeworld: {homeworld}</h3>
      <h3>Population: {population}</h3>
      <h3>Species: {species}</h3>
      <ul>Films: {allFilms}</ul>
      <div className='div-favorite'>
        <button id={name} onClick={() => {toggleFavorite(character)}}>{btnMessage}</button>
      </div>
    </div>
  )
}

export default CharacterCard;

CharacterCard.propTypes = {
  character: PropTypes.object,
  toggleFavorite: PropTypes.func,
  favorites: PropTypes.array
}