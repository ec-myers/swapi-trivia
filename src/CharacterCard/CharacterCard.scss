$accent-color-dark: rgba(108, 91, 117, 0.8);
$accent-color-light: rgba(242,136,79, 0.8);
$hover-color: rgba(231,101,81, 0.8);
$font-color: #ffffff;

.Character-Card {
  background-color: $accent-color-dark;
  border-radius: 10px;
  border: solid 2px $hover-color;
  box-shadow: 0 0 5px #ffffff;
  box-sizing: border-box;
  color: $font-color;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 20px;

  h2 {
    border-bottom: 4px solid #ffffff;
    font-size: 28px;
    padding-bottom: 5px;
    text-align: center;
  }

  h3 {
    font-size: 24px;
    margin: 9px;
  }

  ul {
    font-size: 24px;
    font-weight: bold;
    list-style-type: none;
    margin: 9px;
    min-height: 250px;
  }

  li {
    font-weight: normal;
    margin: 10px;
  }

  .div-favorite {
    text-align: center;
  }

  button {
    background: $accent-color-light;
    border-radius: 5px;
    color: $font-color;
    font-size: 18px;
    height: 40px;
    margin: 10px;
    width: 225px;

    &:hover {
      cursor: pointer;
      background-color: $hover-color;
    }
  }
}

.favorite {
  border: 3px solid white;
  background: rgba(108, 91, 117, 0.9);

  button {
    background: $hover-color;
    &:hover {
      background-color: $accent-color-light;
    }
  }

}