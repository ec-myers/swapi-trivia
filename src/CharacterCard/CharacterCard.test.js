import React from 'react';
import { shallow } from  'enzyme';
import CharacterCard from './CharacterCard';

describe('CharacterCard', () => {
  let wrapper;
  let mockToggleFavorites = jest.fn();
  
  let character = {
    name: 'Luke Skywalker',
    homeworld: 'Tattooine',
    population: 200000,
    species: 'human',
    films: ['blah', 'blah']
  };

  let favorites = [character];

  beforeEach(() => {
    wrapper = shallow(
      <CharacterCard
      key={character.name}
      character={character}
      toggleFavorite={mockToggleFavorites}
      favorites={favorites}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should favorite/unfavorite a character when clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockToggleFavorites).toHaveBeenCalled();
  });
});