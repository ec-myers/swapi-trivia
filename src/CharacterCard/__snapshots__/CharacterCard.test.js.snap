// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match the snapshot 1`] = `
<div
  className="Character-Card favorite"
>
  <h2>
    Luke Skywalker
  </h2>
  <h3>
    Homeworld: 
    Tattooine
  </h3>
  <h3>
    Population: 
    200000
  </h3>
  <h3>
    Species: 
    human
  </h3>
  <ul>
    Films: 
    <li
      key="blah"
    >
      blah
    </li>
    <li
      key="blah"
    >
      blah
    </li>
  </ul>
  <div
    className="div-favorite"
  >
    <button
      id="Luke Skywalker"
      onClick={[Function]}
    >
      Remove Favorite
    </button>
  </div>
</div>
`;
